src/Features/CSharp/Portable/UsePatternMatching/CSharpIsAndCastCheckWithoutNameDiagnosticAnalyzer.cs

```diff
@@ -54,9 +53,17 @@ public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
             => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
 
         protected override void InitializeWorker(AnalysisContext context)
-            => context.RegisterSyntaxNodeAction(SyntaxNodeAction, SyntaxKind.IsExpression);
+        {
+            context.RegisterCompilationStartAction(context =>
+            {
+                var expressionTypeOpt = context.Compilation.GetTypeByMetadataName("System.Linq.Expressions.Expression`1");
```

Comment:
nit: Is this method available to use here?

https://github.com/dotnet/roslyn/blob/cb400a925dc235b505ee2163dc1501d3433d2a12/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ICompilationExtensions.cs#L101-L102

Reply:
yup.